Sources/Templates/TemplateIO.c

```diff
@@ -0,0 +1,298 @@
+#include "General/Calc.h"
+#include "Templates/TemplateIO.h"
+
+#include <assert.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+// Format (all numbers are big-endian):
+// 4B magic "FMR\0"
+// 4B version (ignored, set to " 20\0"
+// 4B total length (including header)
+// 2B rubbish (zeroed)
+// 2B image size in pixels X
+// 2B image size in pixels Y
+// 2B rubbish (pixels per cm X, set to 196 = 500dpi)
+// 2B rubbish (pixels per cm Y, set to 196 = 500dpi)
+// 1B rubbish (number of fingerprints, set to 1)
+// 1B rubbish (zeroed)
+// 1B rubbish (finger position, zeroed)
+// 1B rubbish (zeroed)
+// 1B rubbish (fingerprint quality, set to 100)
+// 1B minutia count
+// N*6B minutiae
+//      2B minutia position X in pixels
+//          2b (upper) minutia type (01 ending, 10 bifurcation, 00 other)
+//      2B minutia position Y in pixels (upper 2b ignored, zeroed)
+//      1B direction, compatible with SourceAFIS angles
+//      1B quality (ignored, zeroed)
+// 2B rubbish (extra data length, zeroed)
+// N*1B rubbish (extra data)
+void TemplateIO_ISO19794_2_2005_Export(Template *template, const char *outputFileName)

```

Comment:
I am happy with asserts in the test code, but not in the library, to me the library code should be bullet-proof no way it can go wrong. It can fail to load a file or save a file, but if it fails all is good nothing bad happens, bar the file not being saved.